### PR TITLE
fix mac unitest on ci

### PR DESCRIPTION
**Pull request by JiabinYang**

### CODE REVIEW

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -168,9 +175,9 @@ EOF
         -DWITH_CONTRIB=${WITH_CONTRIB:-ON} \
         -DWITH_INFERENCE=${WITH_INFERENCE:-ON} \
         -DWITH_INFERENCE_API_TEST=${WITH_INFERENCE_API_TEST:-ON} \
-        -DINFERENCE_DEMO_INSTALL_DIR=${INFERENCE_DEMO_INSTALL_DIR:-/root/.cache/inference_demo} \
```
Comment (typhoonzero) [inline paddle/scripts/paddle_build.sh:171]: add line `-DINFERENCE_DEMO_INSTALL_DIR=${INFERENCE_DEMO_INSTALL_DIR}`

Comment (JiabinYang) [reply paddle/scripts/paddle_build.sh:171]: fixed

**Review by typhoonzero — DISMISSED** [review]
LGTM

File: paddle/scripts/paddle_build.sh (hunk 1 above)
Comment (typhoonzero) [reply paddle/scripts/paddle_build.sh:171]: Why removed again? this is the settings actually used, above are log printing.

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -108,7 +108,15 @@ function cmake_gen() {
...
         fi
     fi
-
+    
+    if [ "$SYSTEM" == "Darwin" ]; then
+        WITH_DISTRIBUTE=${WITH_DISTRIBUTE:-ON}
+        WITH_AVX=${WITH_AVX:-ON}
+        INFERENCE_DEMO_INSTALL_DIR=${INFERENCE_DEMO_INSTALL_DIR}
+    else
+        INFERENCE_DEMO_INSTALL_DIR=${INFERENCE_DEMO_INSTALL_DIR=-/root/.cache/inference_demo}
```
Comment (typhoonzero) [inline paddle/scripts/paddle_build.sh]: should be `INFERENCE_DEMO_INSTALL_DIR=${INFERENCE_DEMO_INSTALL_DIR:-/root/.cache/inference_demo}`

`= -> :`

File: paddle/scripts/paddle_build.sh [hunk 3]
```diff
@@ -108,7 +108,15 @@ function cmake_gen() {
            fi
         fi
     fi
-
+    
+    if [ "$SYSTEM" == "Darwin" ]; then
+        WITH_DISTRIBUTE=${WITH_DISTRIBUTE:-ON}
+        WITH_AVX=${WITH_AVX:-ON}
+        INFERENCE_DEMO_INSTALL_DIR=${INFERENCE_DEMO_INSTALL_DIR}
```
Comment (typhoonzero) [inline paddle/scripts/paddle_build.sh]: seems need to set to `INFERENCE_DEMO_INSTALL_DIR=${INFERENCE_DEMO_INSTALL_DIR:-~/.cache/inference_demo}`

File: paddle/scripts/paddle_build.sh (hunk 1 above)
Comment (JiabinYang) [reply paddle/scripts/paddle_build.sh]: when I add this line it causes the unittest of inference failed due to wrong path of test dataset

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**